test/powershell/Provider/Pester.AutomountedDrives.Tests.ps1

```diff
@@ -0,0 +1,55 @@
+﻿<############################################################################################ 
+ # File: Pester.AutomountedDrives.Tests.ps1
+ # Pester.AutomountedDrives.Tests suite contains Tests that are
+ # used for validating automounted PowerShell drives.
+ ############################################################################################>
+$script:TestSourceRoot = $PSScriptRoot
+Describe "Test suite for validating automounted PowerShell drives" -Tags "CI","Slow" {
+
+    BeforeAll {
+        $powershell = Join-Path -Path $PsHome -ChildPath "powershell"
+        
+        $AutomountVHDDriveScriptPath = Join-Path $script:TestSourceRoot 'AutomountVHDDrive.ps1'
+        $vhdPath = Join-Path $TestDrive 'TestAutomountVHD.vhd'
+
+        $AutomountSubstDriveScriptPath = Join-Path $script:TestSourceRoot 'AutomountSubstDrive.ps1'
+        $substDir = Join-Path (Join-Path $TestDrive 'TestAutomountSubstDrive') 'TestDriveRoot'
+        New-Item $substDir -ItemType Directory -Force | Out-Null
+
+        $SubstNotFound = $false
+        try { subst.exe } catch { $SubstNotFound = $true }
+
+        $VHDToolsNotFound = $false
+        try
+        {
+            $tmpVhdPath = Join-Path $TestDrive 'TestVHD.vhd'
+            New-VHD -path $tmpVhdPath -SizeBytes 5mb -Dynamic -ErrorAction Stop
+            Remove-Item $tmpVhdPath
+        }
+        catch
+        { $VHDToolsNotFound = $true }
+    }
+    
+    Context "Validating automounting FileSystem drives" {
+        
+        It "Test automounting using subst.exe" -Skip:$SubstNotFound {
+           & $powershell -noprofile -command "$AutomountSubstDriveScriptPath -FullPath $substDir" | Should Be "Drive found"

```

Comment:
This could fail in a couple ways if there are spaces in the path, a fix would look something like this:

You can use:
```powershell
& $powershell -noprofile -command "& '$AutomountSubstDriveScriptPath' -FullPath '$substDir'" | Should Be "Drive found"
```